js/index.js

```diff
@@ -1,42 +1,68 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  //console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span').innerHTML;
+  const quantity = product.querySelector('.quantity input').value;
+  const subtotalPrice = +price * quantity;
+  const subtotal = product.querySelector('.subtotal span');
+  subtotal.innerHTML = subtotalPrice;
+  return subtotalPrice;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
-  // ITERATION 3
-  //... your code goes here
+  const products = document.getElementsByClassName('product');
+  let totalPrice = 0;
+  for (let product of products) {
+    totalPrice += updateSubtotal(product);
+  }
+  const total = document.querySelector('#total-value span');
+  total.innerHTML = totalPrice;
```

Comment:
```suggestion
  total.innerText = totalPrice;
```
Lo mismo que antes :D